_posts/2016-10-26-building-net-core-on-travis-ci.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: "Building .Net Core On Travis CI"
+date: 2016-10-26 13:20:49
+tags: 
+- .NET Core
+- CI
+- Travis CI
+categories: Development
+twitter_text: "Building #aspnetcore on @travisci"
+authors:
+- Khalid Abuhakmeh
+image: https://farm4.staticflickr.com/3831/14120933297_089ce68cf7_o_d.jpg
+image_url: https://www.flickr.com/photos/mvegmond/
+image_credit: kweetniet2012
+---
+
+[.NET Core][dotnet] is a cross-platform runtime, and if you play your developer cards right, you can take advantage of running your application on Windows, macOS, and Linux. We enjoy writing many open source projects. Sometimes we opt to use our internal TeamCity CI server and other times we use [Travis CI][travis] for more transparency.  This post will help you get your .NET Core application building in Travis CI by showing you the files you need in your publicly accessible Git repository.
+
+## Build.sh
+
+You need a build script to the root of our repository for Travis CI to execute. It is quite simple.
+
+```
+#!/usr/bin/env bash
+dotnet restore && dotnet build **/project.json
+```
+
+You can add additional steps to the `build.sh` to run tests, but this is the simplest command you need.
+
+## .Travis.yml 
+
+The `.travis.yml` is used by the service to construct your build environment. Add this to the root of your repository.
+
+```
+language: csharp
+dist: trusty
+sudo: required
+mono: none
+dotnet: 1.0.0-preview2-003121
+script:
+  - ./build.sh --quiet verify
+``` 
+
+We do a few things in the configuration file:
+
+- set the language to `csharp`
+- set the linux distribution to `trusty`, a newer version of Ubuntu
```

Comment:
Trusty is 14.04. 16.04 was released back in April of this year, but TravisCI does not support it yet. And, their default dist. is 12.04, which is not supported by the .NET Core installer.